cognite/client/_api/assets.py

```diff
@@ -427,8 +432,188 @@ def aggregate_metadata_values(
+        return self._aggregate2(
+            "count",
+            properties=property,
+            filter=filter,
+            advanced_filter=advanced_filter,
+        )
+
+    def aggregate_cardinality(
+        self,
+        property: AssetPropertyLike | tuple[AssetPropertyLike, AggregationFilter],
+        advanced_filter: Filter | dict | None = None,
+        aggregate_filter: AggregationFilter | dict | None = None,
+        filter: AssetFilter | dict | None = None,
+    ) -> int:
+        """`Find approximate count of asset properties. <https://developer.cognite.com/api#tag/Assets/operation/aggregateAssets>`_
+
+        Args:
+            property (AssetPropertyLike | tuple[AssetPropertyLike, AggregationFilter]): The property to count the cardinality of.
+            advanced_filter (Filter | dict | None): The filter to narrow down the assets to count cardinality.
+            aggregate_filter (AggregationFilter | dict | None): The filter to apply to the resulting buckets.
+            filter (AssetFilter | dict | None): The filter to narrow down the assets to count requirering exact match.
+        Returns:
+            int: The number of properties matching the specified filters and search.
+
+        Examples:
+
+        Count the number of labels used by assets in your CDF project:
+
+            >>> from cognite.client import CogniteClient
+            >>> from cognite.client.data_classes.assets import AssetProperty
+            >>> c = CogniteClient()
+            >>> label_count = c.assets.aggregate_cardinality(AssetProperty.labels)
+
+        Count the number of timezones (metadata key) for assets with the word "critical" in the description
+        in your CDF project:
+
+            >>> from cognite.client import CogniteClient
+            >>> from cognite.client.data_classes import filters
+            >>> from cognite.client.data_classes.assets import AssetProperty
+            >>> c = CogniteClient()
+            >>> is_critical = filters.Search(AssetProperty.description, "critical")
+            >>> critical_assets = c.assets.aggregate_cardinality(AssetProperty.metadata_key("timezone"), advanced_filter=is_critical)
+
+        """
+        self._validate_filter(advanced_filter)
+
+        if property == ["metadata"] or property is AssetProperty.metadata:
+            if isinstance(property, tuple):
+                property = property[0]
+
+            return self._aggregate2(
+                "cardinalityProperties",
+                path=property,
+                filter=filter,
+                advanced_filter=advanced_filter,
+                aggregate_filter=aggregate_filter,
+            )
+        else:
+            return self._aggregate2(
+                "cardinalityValues",
+                properties=property,
+                filter=filter,
+                advanced_filter=advanced_filter,
+                aggregate_filter=aggregate_filter,
+            )
+
+    def aggregate_unique(
```

Comment:
These need to be split into `aggregate_unique_values` and `aggregate_unique_properties`. Same for cardinality aggregates. Should be able to aggregate values for metadata fields.